### PR TITLE
fix(wdio-cucumber-framework): add cucumber 6 retries support and scenario

### PR DESCRIPTION
**Pull request by alfonso-presa**
… level reporting

## Proposed changes

Enable scenario level reporting and some minor fixes to allow cucumber v6 retry feature to work.

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [X] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: packages/wdio-cucumber-framework/src/reporter.js [hunk 1]
```diff
@@ -133,27 +141,42 @@ class CucumberReporter {
...
+            this.emit('suite:end', {
+                uid: getUniqueIdentifier(scenario, sourceLocation),
+                title: this.getTitle(scenario),
+                parent: getUniqueIdentifier(feature),
+                type: 'scenario',
+                file: uri,
+                duration: new Date() - this.scenarioStart,
+                tags: scenario.tags
+            })
+        }
```
Comment (christian-bromann) [inline packages/wdio-cucumber-framework/src/reporter.js:179]:
```suggestion
        if(this.scenarioLevelReport) {
            return this.afterTest(uri, feature, scenario, undefined, result, sourceLocation)
        }
        
        this.emit('suite:end', {
            uid: getUniqueIdentifier(scenario, sourceLocation),
            title: this.getTitle(scenario),
            parent: getUniqueIdentifier(feature),
            type: 'scenario',
            file: uri,
            duration: new Date() - this.scenarioStart,
            tags: scenario.tags
        })
```

File: docs/ConfigurationFile.md [hunk 2]
```diff
@@ -227,6 +227,7 @@ exports.config = {
         tagExpression: [],  // <string[]> (expression) only execute the features or scenarios with tags matching the expression
         timeout: 20000,     // <number> timeout for step definitions
         ignoreUndefinedDefinitions: false, // <boolean> Enable this config to treat undefined definitions as warnings.
+        scenarioLevelReporter: false // Enable this to make webdriver.io behave as if scenarios and not steps were the tests.
```
Comment (christian-bromann) [inline docs/ConfigurationFile.md:230]: This property should also be documented in https://github.com/webdriverio/webdriverio/blob/master/packages/wdio-cucumber-framework/cucumber-framework.d.ts#L111

**Comment by alfonso-presa** [comment]
@christian-bromann I had it almost working without major changes with retry (it worked only by not shifting the `acceptedPickles`), but I noted that the report was not working correctly when I wrote the acceptance tests as it was not thinking about retried scenarios, but about retried suites, and it was failing the run even if the retried test was success. So basically, what drove me to add  scenarios tests was making the smoke test pass :-S. I tried to do it with as much sense as I could. To be honest IMHO it's a more natural behaviour to have test<->scenario correlation than test<->step one, but I understand and respect people thinking the other way arround.

File: docs/ConfigurationFile.md (hunk 2 above)
Comment (alfonso-presa) [reply docs/ConfigurationFile.md:230]: Right away! ᕕ( ᐛ )ᕗ

**Pull request merged**